capstone-frontend/src/components/SearchUserModal.jsx

```diff
@@ -0,0 +1,256 @@
+
+    const indexAddedUsers = this.state.addedUsers.indexOf(user);
+    this.state.addedUsers.splice(indexAddedUsers, 1)
+
+    if (!this.state.addedFriends.includes(user)) {
+      const indexaddedUsersTracker = this.state.addedUsersTracker.indexOf(user);
+      this.state.addedUsersTracker.splice(indexaddedUsersTracker, 1)
+    }
+
+    // Rerender
+    this.setState({ addedUsers: this.state.addedUsers, addedUsersTracker: this.state.addedUsersTracker })
+
+  }
+
+  removeUserFromAddedFriends = (user) => {
+
+    const index = this.state.addedFriends.indexOf(user);
+    this.state.addedFriends.splice(index, 1)
+
+    if (!this.state.addedUsers.includes(user)) {
+      const indexaddedUsersTracker = this.state.addedUsersTracker.indexOf(user);
+      this.state.addedUsersTracker.splice(indexaddedUsersTracker, 1)
+    }
+
+    // Rerender
+    this.setState({ addedFriends: this.state.addedFriends })
+  }
+
+
+
+  render() {
+    return (
+      <div>
+        <button className={this.props.btnStyle} onClick={() => this.setState({ showModal: true })}>
+          <div className={this.props.textStyle}>
+            <span id='create-list-modal'> {this.props.text || 'Search for Users'} </span>
+          </div>
+        </button>
+
+        <Modal
+          dialogClassName="modal-style"
+          size="lg"
+          show={this.state.showModal}
+          onHide={() => this.setState({ showModal: false, searchTerm: '', searchResults: [], addedUsersTracker:[], addedUsers: [], addedFriends: [] })}
+          aria-labelledby='search-users-modal'>
+
+          <Modal.Header closeButton>
+            <Modal.Title id='search-users-modal'>
+              {this.props.text || 'Search for Users'}
+            </Modal.Title>
+          </Modal.Header>
+          <Modal.Body>
+            <Form>
+              <Form.Group controlId='form-search-term'>
+                <Form.Control type='text' placeholder='Search' onChange={(event) => this.handleSearchTermChange(event)} />
+                {this.state.fetchingUsers &&
+                  <div className='text-center'>
+                    <Spinner
+                      as='span'
+                      animation='border'
+                      size='lg'
+                      role='status'
+                      aria-hidden='true'
+                      className='my-5'
+                    />
+                  </div>}
+              </Form.Group>
+
+              {
+                this.state.searchResults &&
+
+                <div>
+
+                  {this.state.searchResults.length > 0 &&
+                    <h3 className='my-4 px-4'>Search Results</h3>
+                  }
+
+                  <Row className='px-3 text-center'>
+                    {(!this.state.resultsFound && this.state.searchTerm !== '') &&
+                      <h4 className='margin-auto py-4'>No Users Found</h4>
+                    }
+
+                    {this.state.searchResults.map(user =>
```

Comment:
Could these user cards be abstracted and used for my User Search page too?

Reply:
I am not sure. Probably they are not that proprietary to this modal.